main.nf

```diff
@@ -14,47 +16,111 @@ nextflow.enable.dsl = 2
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    GENOME PARAMETER VALUES
+    IMPORT NF-CORE UTILITY FUNCTIONS / SUBWORKFLOWS
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-params.fasta = WorkflowMain.getGenomeAttribute(params, 'fasta')
+include { PIPELINE_INITIALISATION } from './subworkflows/local/utils_nfcore_oncoanalyser_pipeline'
+include { PIPELINE_COMPLETION     } from './subworkflows/local/utils_nfcore_oncoanalyser_pipeline'
+
+include { getGenomeAttribute } from './subworkflows/local/utils_nfcore_oncoanalyser_pipeline'
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    VALIDATE & PRINT PARAMETER SUMMARY
+    SET DEFAULT VALUES
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-WorkflowMain.initialise(workflow, params, log)
+params.ref_data_genome_fasta         = getGenomeAttribute('fasta')
+params.ref_data_genome_fai           = getGenomeAttribute('fai')
+params.ref_data_genome_dict          = getGenomeAttribute('dict')
+params.ref_data_genome_bwamem2_index = getGenomeAttribute('bwamem2_index')
+params.ref_data_genome_gridss_index  = getGenomeAttribute('gridss_index')
+params.ref_data_genome_star_index    = getGenomeAttribute('star_index')
+
+WorkflowMain.setParamsDefaults(params, log)
+WorkflowMain.validateParams(params, log)
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    NAMED WORKFLOW FOR PIPELINE
+    CREATE PLACEHOLDER FILES FOR STUB RUNS
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-include { ONCOANALYSER } from './workflows/oncoanalyser'
+// NOTE(SW): required prior to workflow import
+
+if (workflow.stubRun && params.create_stub_placeholders) {
+    Utils.createStubPlaceholders(params)
+}
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    IMPORT WORKFLOWS
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
+
+include { TARGETED } from './workflows/targeted'
+include { WGTS     } from './workflows/wgts'
+
+/*
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    NAMED WORKFLOWS FOR PIPELINE
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+*/
 
 //
-// WORKFLOW: Run main nf-core/oncoanalyser analysis pipeline
+// WORKFLOW: Run main analysis pipeline depending on type of input
 //
+run_mode = Utils.getRunMode(params.mode, log)
+
 workflow NFCORE_ONCOANALYSER {
-    ONCOANALYSER ()
+
+    if (run_mode === Constants.RunMode.WGTS) {
+        WGTS()
+    } else if (run_mode === Constants.RunMode.TARGETED) {
+        TARGETED()
+    } else {
+        assert false
```

Comment:
You're right that the processor should never execute that branch and in the case that it does something has gone very wrong, so the assert False exists there to catch and halt the pipeline in that hypothetical. Continuing to try to execute may have detrimental effects.

A better way to handle this would be to exit the workflow gracefully with a message to the user indicating an incorrect mode had been somehow set. That would give them a meaningful error in such a scenario and provide an extra piece of reportable information.

<br />

- I've made the described change in https://github.com/nf-core/oncoanalyser/pull/67